weblab/entities/tests/test_views.py

```diff
@@ -84,11 +84,75 @@ def test_create_protocol_requires_permissions(self, logged_in_user, client):
         assert response.status_code == 302
         assert '/login/' in response.url
 
+@pytest.mark.django_db
+class TestEntityRenaming:
+    def test_model_renaming_success(self, client, logged_in_user, helpers):
+        helpers.add_permission(logged_in_user, 'create_model')
+        model = recipes.model.make(author=logged_in_user)
+        assert model.name == 'my model1'
+
+        response = client.post(
+            '/entities/models/%d/rename' % model.pk,
+            data={
+                     'name': 'new name'
+                 })
+        assert response.status_code == 302
+        entity = ModelEntity.objects.first()
+        assert entity.name == 'new name'
+
+    def test_model_renaming_failure(self, client, logged_in_user, helpers):
+        helpers.add_permission(logged_in_user, 'create_model')
+        model = recipes.model.make(author=logged_in_user)
+
+        model2 = recipes.model.make(name='test model 2')
```

Comment:
Yeah, following from the comment above: this shouldn't fail, because `model2` has a different author from `model`.